engine.js

```diff
@@ -7,8 +7,30 @@ var hmdajson = require('./lib/hmdajson'),
     _ = require('underscore'),
     brijSpec = require('brij-spec'),
     stream = require('stream'),
-    request = require('sync-request'),
-    moment = require('moment');
+    request = require('then-request'),
+    moment = require('moment'),
+    Q = require('q');
+
+var randomResultName = function() {
+    var text = '';
+    var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz';
+
+    for (var i=0; i < 20; i++) {
+        text += possible.charAt(Math.floor(Math.random() * possible.length));
+    }
+    return text;
+};
+
+var iterateArray = function(arr) {
+    var iter = '';
+    for (var i=0; i < arr.length; i++) {
+        iter += arr[i];
+        if (i !== arr.length-1) {
+            iter += ', ';
+        }
+    }
+    return iter;
+};
```

Comment:
Can't this be replaced by `arr.join(', ')`? Also, iterateArray() seems like a poor naming choice for a function that converts an array to a comma separated string.